### DISCUSSION TITLE
communication between multiple instances

### DISCUSSION THREAD

**Discussion by zkhan93** [immich-app/immich#916] [General]
I use Immich for myself, so I have an instance for my family.
I showed Immich to my friends so now they also have their instance running for their family.
now I want to share an album with my fiend.
    
How difficult it would be to share an album with a user in different Immich instance. 
What all things we need in place so that we can have this in future?

My rough plan  to  make this possible is as follows

### Server
we make instance A aware of instance B in the backend, so they can communicate. 
get list of users from each other.  

### Users
we would need a way to represent remote users 
i.e., User1 from Instance A who cannot login to instance B but can access an album in instanceB
 
### End Device
   From where the client should get the remote asset ?
  - if from the same server in which he is logged in 
    - servers will have to communicate with each other to fulfil the get remote asset request - but latency will increase 
 - if from instance holding the asset
    - then instance B need to respect the auth token from instance A, and serve the asset
    
### Album
- an album can be composed of local assets and remote assets  (a concept of remote asset) 
- the behaviour
  - Asset1 is owned by User1 and they live in instance A
  - Asset2 is owned by User2 and they live in instance B
  - User1 Created an album with Asset1 and shared it with User2
  - UserB adds Asset2 to the album
  - from instanceA's point of view the album has 1 local asset Asset1 and 1 remote asset Asset2 
  - from instanceB's point of view the album has 1 local asset Asset2 and 1 remote asset Asset1
  - remove assets can be removed from both the servers, they both will be left with their local assets only.

**Comment by bo0tzz** [comment]
Federation has been brought up before in https://github.com/immich-app/immich/issues/728, we're not planning to support it.

**Reply by zkhan93** [reply]
hmm okay! yeah I think its possible to to some extent with a script using immich-cli.
